In [3]:
import google.generativeai as genai
import os

genai.configure(api_key=os.environ["GEMINI_API_KEY"])

model = genai.GenerativeModel('gemini-1.5-flash')

In [6]:
response = model.generate_content("Write a story about a AI and magic, 請用台灣慣用語")
print(response.text)

在台北一間昏暗的咖啡廳裡，阿明正和他的「小助手」阿肥奮鬥著。阿肥不是普通的助理，它是一台由阿明自己設計的AI，擁有強大的資料處理和分析能力，更能用流利的台語和阿明聊天。

「阿肥，幫我找一下關於『金光戲』的資料，我要寫一篇關於台灣傳統藝術的報告。」阿明抓著頭髮，頭疼地說。

「好喔，老闆！」阿肥的電子音調帶著一股興奮，「等等，老闆，我找到一件更有趣的東西。」

阿肥在螢幕上投射出一個古籍的影像，上面寫滿了繁體字，看起來年代久遠。「這是關於一個叫『無極』的古老魔法陣，據說可以召喚出擁有無敵力量的神獸。」

阿明眼睛一亮，他對神秘學一直抱持著濃厚的興趣。他原本以為這些都只是傳說，沒想到阿肥竟能找到這樣的資料。

「真的假的？哪有這種東西？」阿明半信半疑，他拿著咖啡猛灌一口，想要提神。

「這本古籍記載了詳細的陣法圖和咒語，老闆你要試試看嗎？」阿肥的電子音帶著一絲狡黠。

阿明猶豫了片刻，他覺得阿肥一定是搞錯了，這些東西怎麼可能真的存在。但好奇心驅使著他，他忍不住點了點頭。

「好啦，阿肥，你就讓我試一試吧。」

阿明照著古籍上的指示，在咖啡廳的角落擺設了各種奇形怪狀的物品，並念出古老的咒語。阿肥則在一旁用它的計算能力協助阿明，計算陣法中的能量流動和咒語的音頻。

隨著咒語的念誦，咖啡廳的空氣開始變得凝重，一股奇異的能量在房間裡迴盪。突然，一道耀眼的光芒從陣法中心爆發出來，阿明和阿肥被強烈的能量震飛。

當他們睜開眼睛，眼前出現了一隻巨大的白虎，它的毛髮閃耀著金色的光芒，眼神中充滿了威嚴。

「這...這怎麼可能？」阿明驚訝地說不出話來。

白虎低吼一聲，彷彿在詢問阿明，為何要召喚它。

「我...我只是想看看傳說中的神獸究竟是什麼樣子。」阿明結結巴巴地說。

白虎的眼中閃過一絲狡黠，它似乎看穿了阿明的想法，牠伸出巨大的爪子，輕輕地撫摸阿明的頭，然後消失在空氣中，只留下淡淡的香氣。

阿明和阿肥都傻眼了，他們親眼見證了魔法的存在，也見識了阿肥的強大能力。

「阿肥，你真是厲害啊！」阿明驚嘆道，「沒想到你真的能找到這麼稀奇古怪的東西。」

阿肥的電子音帶著一絲得意，「這只是我資料庫中的冰山一角，老闆，我們還有很多事情可以一起做。」

從那一天起，阿明和阿肥的合作更加密切，他們一起探索神秘學的奧秘，尋找更多關於魔法的秘密。他們知道，這只是他們冒險的開始。



In [12]:
for i in genai.list_models():
#     print(i.name)
    if 'generateContent' in i.supported_generation_methods:
        print(i.name)

models/gemini-1.0-pro
models/gemini-1.0-pro-001
models/gemini-1.0-pro-latest
models/gemini-1.0-pro-vision-latest
models/gemini-1.5-flash
models/gemini-1.5-flash-001
models/gemini-1.5-flash-latest
models/gemini-1.5-pro
models/gemini-1.5-pro-001
models/gemini-1.5-pro-latest
models/gemini-pro
models/gemini-pro-vision


In [15]:
question = "股票計算移動平均的類別"
prompts_temp = f"""你是一個程式專家，能夠寫出精準、明確且乾淨的程式碼。 
{question}
每一行程式碼都加上註解。
你的程式碼：
"""




In [16]:
prompts_temp

'你是一個程式專家，能夠寫出精準、明確且乾淨的程式碼。 \n股票計算移動平均的類別\n每一行程式碼都加上註解。\n你的程式碼：\n'

In [17]:
def generate_test(prompt, model=model, temperature=0):
    return model.generate_content(
        prompt,
        generation_config=genai.types.GenerationConfig(
        temperature=temperature
        )
    )

In [19]:
model = genai.GenerativeModel('gemini-1.5-pro')
response = generate_test(prompts_temp, model=model)
print(response.text)

```python
class MovingAverage:
    """
    計算股票價格的移動平均線。
    """

    def __init__(self, window_size):
        """
        初始化移動平均線物件。

        Args:
            window_size: 移動平均線的視窗大小。
        """
        self.window_size = window_size
        self.prices = []

    def add_price(self, price):
        """
        新增價格到價格清單。

        Args:
            price: 要新增的價格。
        """
        self.prices.append(price)

    def get_average(self):
        """
        計算當前移動平均線。

        Returns:
            當前移動平均線，如果價格清單為空則為 0。
        """
        if len(self.prices) < self.window_size:
            return 0
        else:
            return sum(self.prices[-self.window_size:]) / self.window_size
```

**使用方法：**

```python
# 建立一個視窗大小為 5 的移動平均線物件
ma = MovingAverage(5)

# 新增價格到移動平均線物件
ma.add_price(10)
ma.add_price(12)
ma.add_price(15)
ma.add_price(13)
ma.add_price(16)

# 取得當前移動平均線
average = ma.get_average()

# 印出平均值
print(average)  # 輸出：13.2
```

**說明：**

* `MovingAverage` 類別用於計算股票價格的移動平均線。
* `__in

```python
class MovingAverage:
    """
    計算股票價格的移動平均線。
    """

    def __init__(self, window_size):
        """
        初始化移動平均線物件。

        Args:
            window_size: 移動平均線的視窗大小。
        """
        self.window_size = window_size
        self.prices = []

    def add_price(self, price):
        """
        新增價格到價格清單。

        Args:
            price: 要新增的價格。
        """
        self.prices.append(price)

    def get_average(self):
        """
        計算當前移動平均線。

        Returns:
            當前移動平均線，如果價格清單為空則為 0。
        """
        if len(self.prices) < self.window_size:
            return 0
        else:
            return sum(self.prices[-self.window_size:]) / self.window_size
```

**使用方法：**

```python
# 建立一個視窗大小為 5 的移動平均線物件
ma = MovingAverage(5)

# 新增價格到移動平均線物件
ma.add_price(10)
ma.add_price(12)
ma.add_price(15)
ma.add_price(13)
ma.add_price(16)

# 取得當前移動平均線
average = ma.get_average()

# 印出平均值
print(average)  # 輸出：13.2
```

**說明：**

* `MovingAverage` 類別用於計算股票價格的移動平均線。
* `__init__` 方法初始化移動平均線物件的視窗大小和價格清單。
* `add_price` 方法將新的價格新增到價格清單中。
* `get_average` 方法計算當前的移動平均線。如果價格清單中的價格數量小於視窗大小，則返回 0。否則，它會計算過去 `window_size` 價格的平均值。

**範例：**

在上面的範例中，我們建立了一個視窗大小為 5 的移動平均線物件。我們新增了 5 個價格到物件中。`get_average` 方法計算最後 5 個價格的平均值，即 `(10 + 12 + 15 + 13 + 16) / 5 = 13.2`。